src/main/scala/rise/core/IsClosedForm.scala

```diff
@@ -6,62 +6,71 @@ import rise.core.traverse._
 import rise.core.types._
 
 object IsClosedForm {
-  case class Visitor(boundV: Set[Identifier], boundT: Set[Kind.Identifier]) extends Traversal[Option] {
-    override implicit def monad = OptionMonad
+  case class Visitor(boundV: Set[Identifier], boundT: Set[Kind.Identifier])
+    extends PureAccumulatorTraversal[(Set[Identifier], Set[Kind.Identifier])]
+  {
+    override val accumulator = PairMonoid(SetMonoid, SetMonoid)
 
-    override def identifier[I <: Identifier] : VarType => I => Option[I] = {
-      case Reference => i => if (boundV(i)) Some(i) else None
-      case _ => return_
+    override def identifier[I <: Identifier]: VarType => I => Pair[I] = vt => i => {
```

Comment:
What is a `Pair` with a single type variable?

Reply:
This `Pair` type is defined by the `PureAccumulatorTraversal` and pairs the accumulator with another type.